bootstore/src/schemes/v0/fsm.rs

```diff
@@ -0,0 +1,479 @@
+                    pkg: our_pkg.unwrap(),
+                    distributed_shares: BTreeMap::new(),
+                    rack_init_state: Some(RackInitState::Running {
+                        start: self.common.clock,
+                        total_members,
+                        acks: BTreeSet::from([self.common.id.clone()]),
+                        unacked_envelopes: envelopes.clone(),
+                    }),
+                    pending_learn_requests: BTreeMap::new(),
+                }));
+                // Note that while we could only send the envelopes to
+                // connected peers, the network layer will just drop them.
+                // We already have a copy to send on connection stored in
+                // `Rack::InitState::Running.unacked_envelopes`.
+                Output { persist: true, envelopes, api_output: None }
+            }
+            Err(e) => return ApiError::RackInitFailed(e).into(),
+        }
+    }
+
+    /// Initialize a node added after rack initialization
+    pub fn init_learner(&mut self) -> Output {
+        let State::Uninitialized(_) = self.state.as_ref().unwrap() else {
+            return ApiError::PeerAlreadyInitialized.into();
+        };
+
+        let mut state = LearningState { attempt: None };
+        let output = state.new_attempt(&mut self.common);
+        self.state = Some(State::Learning(state).into());
+        output
+    }
+
+    /// This call is triggered locally after RSS runs, in order to retrieve the
+    /// `RackSecret` so that it can be used as input key material.
+    ///
+    /// if the rack secret has not already been loaded, then share retrieval
+    /// will begin.
+    pub fn load_rack_secret(&mut self) -> Output {
+        let (rack_uuid, local_share, threshold, share_digests) = match self
+            .state
+            .as_ref()
+            .unwrap()
+        {
+            // We don't allow retrieval before initialization
+            State::Uninitialized(_) => {
+                return ApiError::RackNotInitialized.into()
+            }
+            State::Learning(_) => return ApiError::StillLearning.into(),
+            State::InitialMember(InitialMemberState { pkg, .. }) => {
+                (pkg.rack_uuid, &pkg.share, pkg.threshold, &pkg.share_digests)
+            }
+            State::Learned(LearnedState { pkg, .. }) => {
+                (pkg.rack_uuid, &pkg.share, pkg.threshold, &pkg.share_digests)
+            }
+        };
+        let expiry =
+            self.common.clock + self.common.config.rack_secret_request_timeout;
+
+        self.common.rack_secret_state.load(
+            rack_uuid,
+            &self.common.peers,
+            &self.common.id,
+            local_share,
+            expiry,
+            threshold.into(),
+            share_digests,
+        )
+    }
+
+    /// An abstraction of a timer tick.
+    ///
+    /// Ticks mutate state and can result in message retries.
```

Comment:
This is no longer true in our model. Ticks do not trigger retries. This only happens on connections, since we are modeling TCP.

Reply:
fixed.